src/components/FiltersModal.tsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import Modal from "react-modal";
+
+Modal.setAppElement("#root");
+
+type TFiltersModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  localFilters: {
+    isOnView: boolean;
+    isPublicDomain: boolean;
+    isHiddenGem: boolean;
+    searchQuery: string;
+  };
+  setLocalFilters: (filters: {
+    isOnView: boolean;
+    isPublicDomain: boolean;
+    isHiddenGem: boolean;
+    searchQuery: string;
+  }) => void;
+  onSubmit: (e: React.FormEvent) => void;
+};
+
+const FiltersModal: React.FC<TFiltersModalProps> = ({
+  isOpen,
+  onClose,
+  localFilters,
+  setLocalFilters,
+  onSubmit,
+}) => {
+  const toggleFilter = (key: keyof typeof localFilters) => {
+    setLocalFilters({
+      ...localFilters,
+      [key]: !localFilters[key],
+    });
+  };
+
+  const handleSearchQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setLocalFilters({
+      ...localFilters,
+      searchQuery: e.target.value,
+    });
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onRequestClose={onClose}
+      contentLabel="Filter Options"
+      className="modal-content"
+      overlayClassName="modal-overlay"
+    >
+      <h2 className="modal-title">Filters & Search</h2>
+      <div className="filters-container">
+        {[
+          ["isHiddenGem", "Hidden gem"],
+          ["isOnView", "On view"],
+          ["iPublicDomain", "Public domain"],
+        ].map(([value, text]) => (
+          <button
+            key={text}
```

Comment:
In this case it doesn't matter because `text` is unique per entry, but usually `value` is a better option to use for `key`, because sometimes labels are the same for different values.